### PR TITLE
PT-552 Use Assets IBlobStorage as Distributed storage for the license file

### PR DESCRIPTION
**Pull request by ksavosteev**
https://virtocommerce.atlassian.net/browse/PT-45

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs [hunk 1]
```diff
@@ -1,26 +1,69 @@
...
-        public LicenseProvider(IOptions<PlatformOptions> platformOptions)
+        public LicenseProvider(IOptions<PlatformOptions> platformOptions, IBlobStorageProvider blobStorageProvider, IBlobUrlResolver blobUrlResolver)
         {
             _platformOptions = platformOptions.Value;
+            _blobStorageProvider = blobStorageProvider;
+            _blobUrlResolver = blobUrlResolver;
         }
+
         public License GetLicense()
+        {
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs:24]: Feel free to change the existing method. There won't be breaking changes.

File: src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs [hunk 2]
```diff
@@ -30,5 +73,27 @@ public License GetLicense()
...
+            using (var stream = _blobStorageProvider.OpenWrite(GetLicenseFileUrl()))
+            {
+                var streamWriter = new StreamWriter(stream);
+                streamWriter.Write(license.RawLicense);
+                streamWriter.Flush();
+            }
+        }
+
+
+        protected string GetLicenseFileUrl()
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs:88]: I think these one-line  methods also can be replaced to using in-place (just note)

**Review by tatarincev — CHANGES_REQUESTED** [review]

**Review by Vectorfield4 — APPROVED** [review]
Please, add verify tests.

File: src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs [hunk 3]
```diff
@@ -1,26 +1,69 @@
...
             License license = null;
 
             var licenseFilePath = Path.GetFullPath(_platformOptions.LicenseFilePath);
-            if (System.IO.File.Exists(licenseFilePath))
+            if (File.Exists(licenseFilePath))
             {
-                var rawLicense = System.IO.File.ReadAllText(licenseFilePath);
-                license = License.Parse(rawLicense, Path.GetFullPath(_platformOptions.LicensePublicKeyPath));
+                var rawLicense = File.ReadAllText(licenseFilePath);
+                license = License.Parse(rawLicense, _platformOptions.LicensePublicKeyResourceName);
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/Licensing/LicenseProvider.cs:66]: Seems like a breaking change. This change should be added to the documentation

**Review by eMazeika — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Pull request merged**